parkings/api/monitoring/export_parkings.py

```diff
@@ -0,0 +1,112 @@
+import csv
+
+from rest_framework import renderers, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from six import StringIO
+
+from ...models import Parking, ParkingCheck
+from .permissions import IsMonitor
+
+
+class CSVRenderer(renderers.BaseRenderer):
+    media_type = "text/csv"
+    format = "csv"
+    render_style = 'binary'
+
+    def render(self, filters, media_type=None, renderer_context=None):
+        time_start = filters["time_start"]
+        time_end = filters["time_end"]
+        operators = filters.get("operators")
+        payment_zones = filters.get("payment_zones")
+        parking_check = filters.get("parking_check")
+
+        kwargs = {
+            "time_start__lte": time_end,
+            "time_end__gte": time_start,
+        }
+
+        if payment_zones:
+            kwargs["zone__code__in"] = payment_zones
+        if operators:
+            kwargs["operator__id__in"] = operators
+
+        parkings = (
+            Parking.objects
+            .filter(**kwargs)
+            .order_by("-time_start")
+            .prefetch_related("operator", "zone")
+        )
+
+        if parking_check:
+            parking_check_parking_ids = (
+                ParkingCheck.objects
+                .filter(found_parking__id__in=parkings)
+                .order_by("found_parking")
+                .values_list("found_parking")
+                .distinct()
+            )
+            parkings = parkings.filter(id__in=parking_check_parking_ids)
```

Comment:
The filtering code here doesn't seem to belong to the CSV renderer, but rather to the ViewSet, or even better they should be implemented as FilterSets. See e.g. parkings/api/enforcement/valid_parking.py for an example of FilterSet.